src/ir/ty.rs

```diff
@@ -1077,7 +1080,7 @@ impl Type {
 
                 if name.is_empty() {
                     let pretty_name = ty.spelling();
-                    if Self::is_valid_identifier(&pretty_name) {
+                    if !Self::is_invalid_name(&pretty_name) {
```

Comment:
Regarding C++:

> Yes, in C++ it's something like (anonymous namespace)::foo. However, I don't think this feature is useful in C++, so it's not a problem.

I kind of disagree, since I've definitely seen similar usage in Gecko, but as long as it doesn't break awfully (it just keeps pre-existing functionality) I think it's fine :).

Could you leave a comment either here or in the `Record` case to explicitly mention it doesn't work in C++ and point to the issue where other approaches were discussed? (those would actually work in C++).